src/wallet/wallet2.cpp

```diff
@@ -1762,8 +1745,6 @@ void wallet2::refresh(uint64_t start_height, uint64_t & blocks_fetched, bool& re
     catch (const std::exception&)
     {
       blocks_fetched += added_blocks;
-      if (pull_thread.joinable())
-        pull_thread.join();
```

Comment:
Why just removed ?

Reply:
Since it's using the pool there's nothing to cleanup here.

Reply:
OK, I found the commit for this, and the deleted-while-live doesn't apply here indeed. Exit is taken care of by the pool implementation since it joins.

Reply:
ah I suppose there should be a tpool.wait() here. ok.